empirical-results.tex

```diff
@@ -1 +1,32 @@
-\section{Empirical Results}
\ No newline at end of file
+\section{Empirical Results}
+
+\newcommand{\rref}[1]{}
+
+In order to evaluate our consumption mechanisms, we implement the federated consumption mechanism in Solidity. We provide a concrete implementation of the \textsf{burn-verifier} contract described in Algorithm~\rref{alg.burn-verifier} as \textsf{BurnedBTC}. We implement the \textsf{Crosschain} parent contract from~\cite{pow-sidechains}. In order to achieve this we implement our own verification methods similar to \rref{alg.verify-event-federation} in our \textsf{Verifier} library. Specifically, we verify transaction data in \textsf{Verifier} similarly to \rref{alg.verify-tx} by making use of the open source bitcoin-spv library~\cite{bitcoin-spv-library}. We verify block connection using a Merkle Mountain Range where each leaf corresponds to a block. The leaf stores the merkle tree root of the merkle tree which collects the transaction id of the transactions included in the block in question.
+
+We put forth the \textsf{BurnVerifier} library which provides a concrete implementation for our berning protocol. Specifically the library implements the \textsf{BurnVerify} function of the protocol. Finally, the federation voting happens in the \textsf{CheckpointRepo} contract, which is initialized with the addresses belonging to the federation members. The members can then proceed to call the \textsf{CheckpointRepo.vote} method to vote on their MMR computed from the source blockchain.
```

Comment:
Unless we show these contracts as pseudocode within the paper, there's little point in citing their names. The reader cannot discern what they do. We should try to give a higher-level picture here.